### PR TITLE
Laboratory work #2, Alyona Gosteva - 21FPL1

### PR DESCRIPTION
**Pull request by Gostaa**

### CODE REVIEW

**Comment by artyomtugaryov** [comment]
Hey @Gostaa! As a reminder, the deadline for the second lab is October 21st and you have less than a week to complete it. Please hurry

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Nothing to review(

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Nothing to review

File: lab_1_keywords_tfidf/target_score.txt [hunk 1]
```diff
@@ -1 +1 @@
-10
\ No newline at end of file
+6
```
[the remark this answers is not in the document]
Comment (artyomtugaryov) [reply lab_1_keywords_tfidf/target_score.txt:1]: !

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Please, fix conflicts

File: lab_2_keywords_cooccurrence/start.py [hunk 2]
```diff
@@ -3,11 +3,10 @@
 """
 
 from pathlib import Path
-from lab_2_keywords_cooccurrence.main import (
-    process_text,
-    get_top_n,
-    load_stop_words
-)
+
+from main import extract_phrases, extract_candidate_keyword_phrases, calculate_frequencies_for_content_words, \
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/start.py:7]: `from lab_2_keywords_cooccurrence.main` ...

File: lab_2_keywords_cooccurrence/main.py [hunk 3]
```diff
@@ -120,10 +183,22 @@ def calculate_cumulative_score_for_candidates(candidate_keyword_phrases: KeyPhra
...
+        return None
+    if not check_list_types(candidate_keyword_phrases, tuple):
         return None
-    return {phrase: sum(word_scores[token] for token in phrase) for phrase in candidate_keyword_phrases}
+
+    phrases_scores_dict = {}
+    for phrase in candidate_keyword_phrases:
+        phrases_scores_dict[phrase] = 0
+        for word in phrase:
+            if word in word_scores:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:197]: reverse the condition and return `None`

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/main.py [hunk 4]
```diff
@@ -30,13 +19,32 @@ def extract_phrases(text: str) -> Optional[Sequence[str]]:
...
-                            r"|(?<=\s)[^\s\w]+"  # punctuation after the whitespace symbol
-                            r"|[^\s\w]+(?=\s)"  # punctuation before the whitespace symbol
-                            r"|[^\s\w]+(?=$)")  # punctuation before the end of line
-    return [clean for phrase in re.split(expression, text) if (clean := phrase.strip())]
+    full_punctuation = punctuation + "–—¡¿⟨⟩«»…⋯‹›“”"
+    for symbol in full_punctuation:
+        text = text.replace(symbol, '.')
+    phrases = text.split('.')
+    phrases_stripped = []
+    for string in phrases:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:29]: do it in one line

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/start.py (hunk 2 above)
Comment (Gostaa) [reply lab_2_keywords_cooccurrence/start.py:7]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 4 above)
Comment (Gostaa) [reply lab_2_keywords_cooccurrence/main.py]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 3 above)
Comment (Gostaa) [reply lab_2_keywords_cooccurrence/main.py]: done

**Review by artyomtugaryov — APPROVED** [review]

**Pull request closed without merging**